### PR TITLE
Remove python 2.x support.

### PR DESCRIPTION
**Pull request by akodali18**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,6 @@
 dist: xenial
 language: python
 python:
-- "2.7"
 - "3.5"
 - "3.6"
 - "3.7"
```
Comment (ustinov) [inline .travis.yml:6]: Please add `3.8` to the list

**Comment by akodali18** [comment]
> please bump the version in `setup.py` as well

@artem any idea how to get around this : https://travis-ci.com/github/wavefrontHQ/wavefront-opentracing-sdk-python/jobs/377802888 ? I believe its complaining since i didn’t increment the version in setup.py (however i don’t want to since the previous release was never made ) - I guess there is no way around this, if there is (other than disabling parts of travis, let me know)

File: .travis.yml [hunk 2]
```diff
@@ -1,7 +1,6 @@
 dist: xenial
```
Comment (ustinov) [inline .travis.yml:1]:
```suggestion
dist: focal
```
Should we also go with the latest LTS?

File: setup.py [hunk 3]
```diff
@@ -32,7 +32,6 @@
         ],
     classifiers=[
         'License :: OSI Approved :: Apache Software License',
-        'Programming Language :: Python :: 2.7',
         'Programming Language :: Python :: 3.5',
         'Programming Language :: Python :: 3.6',
         'Programming Language :: Python :: 3.7'
```
Comment (ustinov) [inline setup.py]:
```suggestion
        'Programming Language :: Python :: 3.7'
        'Programming Language :: Python :: 3.8'
```

Comment (akodali18) [reply setup.py]: Already made this change.

**Comment by ustinov** [comment]
@akodali18 we can also safely delete few `pylint` directives now, like
```
./wavefront_opentracing_sdk/reporting/reporter.py:# pylint: disable=E0012,R0205
./wavefront_opentracing_sdk/sampling/sampler.py:# pylint: disable=E0012,R0205
./wavefront_opentracing_sdk/propagation/registry.py:# pylint: disable=E0012,R0205
```

File: .travis.yml [hunk 4]
```diff
@@ -39,4 +39,3 @@ deploy:
   on:
     python: 3.7
```
Comment (ustinov) [inline .travis.yml:40]:
```suggestion
    python: 3.8
```
Let's also deploy using 3.8

**Comment by akodali18** [comment]
> @akodali18 we can also safely delete few `pylint` directives now, like
> 
> ```
> ./wavefront_opentracing_sdk/reporting/reporter.py:# pylint: disable=E0012,R0205
> ./wavefront_opentracing_sdk/sampling/sampler.py:# pylint: disable=E0012,R0205
> ./wavefront_opentracing_sdk/propagation/registry.py:# pylint: disable=E0012,R0205
> ```

Where do we make this change - to ignore few pylint checks ?

**Comment by ustinov** [comment]
> > @akodali18 we can also safely delete few `pylint` directives now, like
> > ```
> > ./wavefront_opentracing_sdk/reporting/reporter.py:# pylint: disable=E0012,R0205
> > ./wavefront_opentracing_sdk/sampling/sampler.py:# pylint: disable=E0012,R0205
> > ./wavefront_opentracing_sdk/propagation/registry.py:# pylint: disable=E0012,R0205
> > ```
> 
> Where do we make this change - to ignore few pylint checks ?

Those are in the code:
https://github.com/wavefrontHQ/wavefront-opentracing-sdk-python/blob/master/wavefront_opentracing_sdk/propagation/registry.py#L12
https://github.com/wavefrontHQ/wavefront-opentracing-sdk-python/blob/master/wavefront_opentracing_sdk/reporting/reporter.py#L8
https://github.com/wavefrontHQ/wavefront-opentracing-sdk-python/blob/master/wavefront_opentracing_sdk/sampling/sampler.py#L7

We had to inherit explicitly from `(object)` to comply with python2, and python3 does that by default

**Comment by akodali18** [comment]
> > > @akodali18 we can also safely delete few `pylint` directives now, like
> > > ```
> > > ./wavefront_opentracing_sdk/reporting/reporter.py:# pylint: disable=E0012,R0205
> > > ./wavefront_opentracing_sdk/sampling/sampler.py:# pylint: disable=E0012,R0205
> > > ./wavefront_opentracing_sdk/propagation/registry.py:# pylint: disable=E0012,R0205
> > > ```
> > 
> > 
> > Where do we make this change - to ignore few pylint checks ?
> 
> Those are in the code:
> https://github.com/wavefrontHQ/wavefront-opentracing-sdk-python/blob/master/wavefront_opentracing_sdk/propagation/registry.py#L12
> https://github.com/wavefrontHQ/wavefront-opentracing-sdk-python/blob/master/wavefront_opentracing_sdk/reporting/reporter.py#L8
> https://github.com/wavefrontHQ/wavefront-opentracing-sdk-python/blob/master/wavefront_opentracing_sdk/sampling/sampler.py#L7
> 
> We had to inherit explicitly from `(object)` to comply with python2, and python3 does that by default

Thanks done.

File: wavefront_opentracing_sdk/propagation/registry.py [hunk 5]
```diff
@@ -9,8 +9,7 @@
 from . import textmap
 
 
-# pylint: disable=E0012,R0205
-class PropagatorRegistry(object):
+class PropagatorRegistry():
```
Comment (ustinov) [inline wavefront_opentracing_sdk/propagation/registry.py:12]:
```suggestion
class PropagatorRegistry:
```

File: wavefront_opentracing_sdk/reporting/reporter.py [hunk 6]
```diff
@@ -5,8 +5,7 @@
 """
 
 
-# pylint: disable=E0012,R0205
-class Reporter(object):
+class Reporter():
```
Comment (ustinov) [inline wavefront_opentracing_sdk/reporting/reporter.py:8]:
```suggestion
class Reporter:
```

File: wavefront_opentracing_sdk/sampling/sampler.py [hunk 7]
```diff
@@ -4,8 +4,7 @@
 """
 
 
-# pylint: disable=E0012,R0205
-class Sampler(object):
+class Sampler():
```
Comment (ustinov) [inline wavefront_opentracing_sdk/sampling/sampler.py:7]:
```suggestion
class Sampler:
```

**Pull request merged**